<a href="https://colab.research.google.com/github/kitiluck/BSC_DPDM23/blob/main/Freq_HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import pandas as pd
import os

In [6]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [7]:
from google.colab import drive
drive.mount('/content/drive')
path = '/content/drive/MyDrive/BSC_DPDM2023/reduced_marketbasket.csv'

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [8]:
data = pd.read_csv(os.path.join(path,'/content/drive/MyDrive/BSC_DPDM2023/reduced_marketbasket.csv'),encoding='latin-1')
data

InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

             InvoiceDate  UnitPrice  CustomerID         Country  
0       12/01/2010 08:26       2.55     17850.0  United Kingdom  
1       12/01/2010 08:26       3.39     17850.0  United Kingdom  
2       12/01/2010 08:26       2.75     17850.0  United Kingdom  
3       12/01/2010 08:26       3.39     17850.0  United Kingdom  
4       12/01/2010 08:26       3.39     17850.0  United Kingdom  
...                  ...        ...         ...             ...  
541904  12/09/2011 12:50       0.85     12680.0          France  
541905  12/09/2011 12:50       2.10     12680.0          France  
541906  12/09/2011 12:50       4.15     12680.0          France  
541907  12/09/2011 12:50       4.15     12680.0          France  
541908  12/09/2011 12:50       4.95     12680.0          France  

[541909 rows x 8 columns]

In [9]:
data #chek data

InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

             InvoiceDate  UnitPrice  CustomerID         Country  
0       12/01/2010 08:26       2.55     17850.0  United Kingdom  
1       12/01/2010 08:26       3.39     17850.0  United Kingdom  
2       12/01/2010 08:26       2.75     17850.0  United Kingdom  
3       12/01/2010 08:26       3.39     17850.0  United Kingdom  
4       12/01/2010 08:26       3.39     17850.0  United Kingdom  
...                  ...        ...         ...             ...  
541904  12/09/2011 12:50       0.85     12680.0          France  
541905  12/09/2011 12:50       2.10     12680.0          France  
541906  12/09/2011 12:50       4.15     12680.0          France  
541907  12/09/2011 12:50       4.15     12680.0          France  
541908  12/09/2011 12:50       4.95     12680.0          France  

[541909 rows x 8 columns]

In [10]:
data.groupby(['Country']).sum() #ดูข้อมูลมาจากกี่ประเทศ

<ipython-input-10-25f447ef123a>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  data.groupby(['Country']).sum() #ดูข้อมูลมาจากกี่ประเทศ


Quantity    UnitPrice    CustomerID
Country                                                  
Australia                83653     4054.750  1.569300e+07
Austria                   4827     1701.520  5.021102e+06
Bahrain                    260       86.570  2.100270e+05
Belgium                  23152     7540.130  2.571829e+07
Brazil                     356      142.600  4.086080e+05
Canada                    2763      910.580  2.615483e+06
Channel Islands           9479     3738.550  1.128522e+07
Cyprus                    6317     3920.070  7.715880e+06
Czech Republic             592       88.150  3.834300e+05
Denmark                   8188     1266.950  4.876734e+06
EIRE                    142637    48447.190  1.103917e+08
European Community         497      294.050  9.215880e+05
Finland                  10666     3786.850  8.699324e+06
France                  110480    43031.990  1.076489e+08
Germany                 117448    37666.000  1.200751e+08
Greece                    1556      713.290  2.008584e+06
Hong Kong                 4769    12241.500  0.000000e+00
Iceland                   2458      481.210  2.247154e+06
Israel                    4353     1079.040  3.164467e+06
Italy                     7999     3879.390  1.015666e+07
Japan                    25218      814.860  4.567292e+06
Lebanon                    386      242.440  5.743800e+05
Lithuania                  652       99.440  5.366200e+05
Malta                      944      666.010  2.158496e+06
Netherlands             200128     6492.550  3.419054e+07
Norway                   19247     6529.060  1.350765e+07
Poland                    3653     1422.270  4.341972e+06
Portugal                 16180    13037.540  1.886480e+07
RSA                        352      248.100  7.218680e+05
Saudi Arabia                75       24.110  1.256500e+05
Singapore                 5234    25108.890  2.918376e+06
Spain                    26824    12633.450  3.268929e+07
Sweden                   35637     1806.830  6.790083e+06
Switzerland              30325     6813.690  2.377592e+07
USA                       1034      644.980  3.672086e+06
United Arab Emirates       982      229.890  1.018952e+06
United Kingdom         4263829  2245715.474  5.626433e+09
Unspecified               3300     1204.010  3.348046e+06

>>แปลงdata ให้อยู่ในรูปtransaction

In [11]:
data.groupby('InvoiceNo')['StockCode'].apply(list)

InvoiceNo
536365     [85123A, 71053, 84406B, 84029G, 84029E, 22752,...
536366                                        [22633, 22632]
536367     [84879, 22745, 22748, 22749, 22310, 84969, 226...
536368                          [22960, 22913, 22912, 22914]
536369                                               [21756]
                                 ...                        
C581484                                              [23843]
C581490                                       [22178, 23144]
C581499                                                  [M]
C581568                                              [21258]
C581569                                       [84978, 20979]
Name: StockCode, Length: 25900, dtype: object

In [12]:
tarnsac = data.groupby('InvoiceNo')['StockCode'].apply(list)

>>Association Rules

In [13]:
!pip install apyori
from apyori import apriori

  Preparing metadata (setup.py) ... done
  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5954 sha256=8a884ed98d1bdff953eb109e9455fb5cd83bb892f3deb0ae2e075942584a7ab5
  Stored in directory: /root/.cache/pip/wheels/c4/1a/79/20f55c470a50bb3702a8cb7c94d8ada15573538c7f4baebe2d
Successfully built apyori


#Hw1 หา Frequent set item set และ Association rules


*   ของช่วงเวลา Christmas & New year (1 dec - 30 dec)
*   ของประเทศยุโรป
*   ของช่วงเวลาตรุษจีนในทวีป Asia ( 15 Jan - 15 Feb)
*   ของช่วงเวลา Christmas & New year Europe (1 dec - 30 dec)
*   Valentine's day
*   โดยจัดกลุ่มตามประเทศ
groupby ประเทศ - เดือน

In [14]:
import datetime
import pandas as pd
import os
import pandas as pd
from mlxtend.frequent_patterns import apriori
from mlxtend.frequent_patterns import association_rules
from apyori import apriori

In [15]:
data['InvoiceDate'] #chek InvoiceDate

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0         12/01/2010 08:26
1         12/01/2010 08:26
2         12/01/2010 08:26
3         12/01/2010 08:26
4         12/01/2010 08:26
                ...       
541904    12/09/2011 12:50
541905    12/09/2011 12:50
541906    12/09/2011 12:50
541907    12/09/2011 12:50
541908    12/09/2011 12:50
Name: InvoiceDate, Length: 541909, dtype: object

In [16]:
data['InvoiceDate'] = pd.to_datetime(data['InvoiceDate'], format='%m/%d/%Y %H:%M')

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [17]:
data['Date']=data['InvoiceDate'].dt.date

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [18]:
data['Date']=pd.to_datetime(data.Date, format='%Y/%m/%d')

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [19]:
data['Month']=data['InvoiceDate'].dt.month

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [20]:
data

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

## 1. ของช่วงเวลา Christmas & New year (1 dec - 30 dec)

In [21]:
#Select 1-31 Dec
Dec= data[(data['Date'].dt.month == 12) & (data['Date'].dt.day >= 1) & (data['Date'].dt.day <= 31)]

#printcheck
print(Dec)


/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


       InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Dat

In [22]:
Dec

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[68006 rows x 10 columns]

In [23]:
transac1 = Dec.groupby('InvoiceNo')['Description'].apply(list).reset_index()
print(transac1)

     InvoiceNo                                        Description
0       536365  [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       536366  [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       536367  [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       536368  [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4       536369                         [BATH BUILDING BLOCK WORD]
...        ...                                                ...
3035   C581484                      [PAPER CRAFT , LITTLE BIRDIE]
3036   C581490  [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037   C581499                                           [Manual]
3038   C581568                       [VICTORIAN SEWING BOX LARGE]
3039   C581569  [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...

[3040 rows x 2 columns]


/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [24]:
transac1apply = transac1['Description'].apply(lambda x: [str(item) for item in x])
transac1apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0       [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                              [BATH BUILDING BLOCK WORD]
                              ...                        
3035                        [PAPER CRAFT , LITTLE BIRDIE]
3036    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037                                             [Manual]
3038                         [VICTORIAN SEWING BOX LARGE]
3039    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 3040, dtype: object

In [25]:
min_sup = 0.036 #เปลี่ยนจนกว่าจะเจอ...

frequentItemsets = list(apriori(transac1apply,min_support = min_sup))
for i in frequentItemsets:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), support=0.03684210526315789, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), confidence=0.03684210526315789, lift=1.0), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), items_add=frozenset({'ALARM CLOCK BAKELIKE RED '}), confidence=0.7088607594936708, lift=14.462662475575565), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE RED '}), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), confidence=0.7516778523489932, lift=14.462662475575565)])
yes
RelationRecord(items=frozenset({'CHOCOLATE HOT WATER BOTTLE', 'HOT WATER BOTTLE I AM SO POORLY'}), support=0.036513157894736845, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'CHOCOLATE HOT WATER BOTTLE', 'HOT WATER BOTTLE I AM SO POORLY'}), confidence=0.036513157894736845, lift=1.0), O

In [26]:
frequentItemset = list(apriori(transac1apply, min_support=0.036))

for record in frequentItemset:
    if len(record.items) > 1:
        print('yes')
        print(f"Items: {', '.join(record.items)}")
        print(f"Support: {record.support:.4f} ({record.support*100:.2f}%)")

        for i, stat in enumerate(record.ordered_statistics, 1):
            items_base = 'None' if not stat.items_base else ', '.join(stat.items_base)
            items_add = ', '.join(stat.items_add)
            confidence = stat.confidence
            lift = stat.lift

            print(f"Rule {i}: ({items_base} -> {items_add} : Confidence={confidence:.4f}, Lift={lift:.4f})")
        print()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
Items: ALARM CLOCK BAKELIKE GREEN, ALARM CLOCK BAKELIKE RED 
Support: 0.0368 (3.68%)
Rule 1: (None -> ALARM CLOCK BAKELIKE GREEN, ALARM CLOCK BAKELIKE RED  : Confidence=0.0368, Lift=1.0000)
Rule 2: (ALARM CLOCK BAKELIKE GREEN -> ALARM CLOCK BAKELIKE RED  : Confidence=0.7089, Lift=14.4627)
Rule 3: (ALARM CLOCK BAKELIKE RED  -> ALARM CLOCK BAKELIKE GREEN : Confidence=0.7517, Lift=14.4627)

yes
Items: CHOCOLATE HOT WATER BOTTLE, HOT WATER BOTTLE I AM SO POORLY
Support: 0.0365 (3.65%)
Rule 1: (None -> CHOCOLATE HOT WATER BOTTLE, HOT WATER BOTTLE I AM SO POORLY : Confidence=0.0365, Lift=1.0000)
Rule 2: (CHOCOLATE HOT WATER BOTTLE -> HOT WATER BOTTLE I AM SO POORLY : Confidence=0.5163, Lift=9.7484)
Rule 3: (HOT WATER BOTTLE I AM SO POORLY -> CHOCOLATE HOT WATER BOTTLE : Confidence=0.6894, Lift=9.7484)

yes
Items: PAPER CHAIN KIT VINTAGE CHRISTMAS, PAPER CHAIN KIT 50'S CHRISTMAS 
Support: 0.0477 (4.77%)
Rule 1: (None -> PAPER CHAIN KIT 50'S CHRISTMAS , PAPER CHAIN KIT VINTAGE CHRISTMAS : 

มี 3 itemset ที่ผ่าน 0.036
1. คนที่ซื้อทั้ง PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS จาก dataset ทั้งหมด มี 4.77%

   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 4.77%
   
   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 51.6%

   คนที่ซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT 50'S CHRISTMAS มี 66.51%


2. คนที่ซื้อทั้ง ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN จาก dataset ทั้งหมด มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE GREEN แล้วค่อยซื้อ ALARM CLOCK BAKELIKE RED มี 75.17%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED แล้วค่อยซื้อ ALARM CLOCK BAKELIKE GREEN มี 70.89%



3. คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY จาก dataset ทั้งหมด มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE แล้วค่อยซื้อ HOT WATER BOTTLE I AM SO POORLY มี 51.63%

   คนที่ซื้อ HOT WATER BOTTLE I AM SO POORLY แล้วค่อยซื้อ CHOCOLATE HOT WATER BOTTLE มี 68.94%



มี 3 itemset ที่ผ่าน 0.036

RelationRecord(items=frozenset({'22386', '85099B'}),

support=0.03216216216216216,
<t>
<t>

คนที่ซื้อทั้ง 22386, 85099B จาก dataset ทั้งหมดมี 3.21%

ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'22386', '85099B'}), confidence=0.03216216216216216, lift=1.0),

คนที่ซื้อ 22386 แล้วค่อยซื้อ 85099B มีตั้ง 67.66%


OrderedStatistic(items_base=frozenset({'22386'}), items_add=frozenset({'85099B'}), confidence=0.6766856214459789, lift=8.20897311262335),


คนที่ซื้อ 85099B แล้วค่อยซื้อ 22386 มี 39.01%


OrderedStatistic(items_base=frozenset({'85099B'}), items_add=frozenset({'22386'}), confidence=0.39016393442622943, lift=8.20897311262335)])

##2 ของประเทศยุโรป

##จัดกลุ่มประเทศตามทวีป

In [27]:
set(data['Country']) #check country

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


{'Australia',
 'Austria',
 'Bahrain',
 'Belgium',
 'Brazil',
 'Canada',
 'Channel Islands',
 'Cyprus',
 'Czech Republic',
 'Denmark',
 'EIRE',
 'European Community',
 'Finland',
 'France',
 'Germany',
 'Greece',
 'Hong Kong',
 'Iceland',
 'Israel',
 'Italy',
 'Japan',
 'Lebanon',
 'Lithuania',
 'Malta',
 'Netherlands',
 'Norway',
 'Poland',
 'Portugal',
 'RSA',
 'Saudi Arabia',
 'Singapore',
 'Spain',
 'Sweden',
 'Switzerland',
 'USA',
 'United Arab Emirates',
 'United Kingdom',
 'Unspecified'}

In [28]:
Eur = data[(data['Country']!='Bahrain') | (data['Country']!='Hong Kong')| (data['Country']!='Singapore') | (data['Country']!='Japan') | (data['Country']!='Israel') | (data['Country']!='Lebanonl') |
           (data['Country']!='Saudi Arabia') | (data['Country']!='United Arab Emirates')| (data['Country']!='Australia')| (data['Country']!='Brazil') | (data['Country']!='RSA')| (data['Country']!='Canada')|
           (data['Country']!='USA')]
Eur #แบ่งทวีปยุโรป

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

In [29]:
tran_Eur = Eur.groupby('InvoiceNo')['Description'].apply(list).reset_index()
tran_Europe = tran_Eur['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [30]:
tran_Europe

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0        [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1        [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2        [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3        [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                               [BATH BUILDING BLOCK WORD]
                               ...                        
25895                        [PAPER CRAFT , LITTLE BIRDIE]
25896    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
25897                                             [Manual]
25898                         [VICTORIAN SEWING BOX LARGE]
25899    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 25900, dtype: object

In [31]:
min_sup = 0.03 #เปลี่ยนจนกว่าจะเจอ...

frequentItemsets_Europe = list(apriori(tran_Europe, min_support = min_sup))
for i in frequentItemsets_Europe:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'GREEN REGENCY TEACUP AND SAUCER', 'ROSES REGENCY TEACUP AND SAUCER '}), support=0.03027027027027027, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'GREEN REGENCY TEACUP AND SAUCER', 'ROSES REGENCY TEACUP AND SAUCER '}), confidence=0.03027027027027027, lift=1.0), OrderedStatistic(items_base=frozenset({'GREEN REGENCY TEACUP AND SAUCER'}), items_add=frozenset({'ROSES REGENCY TEACUP AND SAUCER '}), confidence=0.7417218543046358, lift=17.1523178807947), OrderedStatistic(items_base=frozenset({'ROSES REGENCY TEACUP AND SAUCER '}), items_add=frozenset({'GREEN REGENCY TEACUP AND SAUCER'}), confidence=0.7, lift=17.1523178807947)])
yes
RelationRecord(items=frozenset({'JUMBO BAG RED RETROSPOT', 'JUMBO BAG PINK POLKADOT'}), support=0.03216216216216216, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'JUMBO BAG PINK POLKADOT', 'JUMBO BAG RED RETROSPOT'}), confidence=0.03216216216216216, lift=1.

In [32]:
frequentItemset_Europe = list(apriori(tran_Europe, min_support=0.03))

for record in frequentItemset_Europe:
    if len(record.items) > 1:
        print('Yes')
        print(f"Items: {', '.join(record.items).lower()}")
        print(f"Support: {record.support:.4f} ({record.support*100:.2f}%)")

        for i, stat in enumerate(record.ordered_statistics, 1):
            items_base = 'None' if not stat.items_base else ', '.join(stat.items_base)
            items_add = ', '.join(stat.items_add)
            confidence = stat.confidence
            lift = stat.lift

            print(f"Rule {i}: ({items_base} -> {items_add} : Confidence={confidence:.4f}, Lift={lift:.4f})")
        print()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Yes
Items: green regency teacup and saucer, roses regency teacup and saucer 
Support: 0.0303 (3.03%)
Rule 1: (None -> GREEN REGENCY TEACUP AND SAUCER, ROSES REGENCY TEACUP AND SAUCER  : Confidence=0.0303, Lift=1.0000)
Rule 2: (GREEN REGENCY TEACUP AND SAUCER -> ROSES REGENCY TEACUP AND SAUCER  : Confidence=0.7417, Lift=17.1523)
Rule 3: (ROSES REGENCY TEACUP AND SAUCER  -> GREEN REGENCY TEACUP AND SAUCER : Confidence=0.7000, Lift=17.1523)

Yes
Items: jumbo bag red retrospot, jumbo bag pink polkadot
Support: 0.0322 (3.22%)
Rule 1: (None -> JUMBO BAG PINK POLKADOT, JUMBO BAG RED RETROSPOT : Confidence=0.0322, Lift=1.0000)
Rule 2: (JUMBO BAG PINK POLKADOT -> JUMBO BAG RED RETROSPOT : Confidence=0.6767, Lift=8.2090)
Rule 3: (JUMBO BAG RED RETROSPOT -> JUMBO BAG PINK POLKADOT : Confidence=0.3902, Lift=8.2090)



มี 2 itemset ที่ผ่าน 0.03

1. คนที่ซื้อทั้ง jumbo bag red retrospot และ jumbo bag pink polkadot จาก dataset ทั้งหมด มี 3.22%

   คนที่ซื้อ JUMBO BAG RED RETROSPOT และ JUMBO BAG PINK POLKADOT มี 3.22%

   คนที่ซื้อ JUMBO BAG PINK POLKADOT แล้วค่อยซื้อ JUMBO BAG RED RETROSPOT มี 67.67%

   คนที่ซื้อ JUMBO BAG RED RETROSPOT แล้วค่อยซื้อ JUMBO BAG PINK POLKADOT มี 39.02%

2. คนที่ซื้อทั้ง green regency teacup and saucer และ roses regency teacup and saucer จาก dataset ทั้งหมด มี 3.03%

   คนที่ซื้อ GREEN REGENCY TEACUP AND SAUCER และ ROSES REGENCY TEACUP AND SAUCER มี 3.03%

   คนที่ซื้อ GREEN REGENCY TEACUP AND SAUCER แล้วค่อยซื้อ ROSES REGENCY TEACUP AND SAUCER มี 74.17%

   คนที่ซื้อ ROSES REGENCY TEACUP AND SAUCER แล้วค่อยซื้อ GREEN REGENCY TEACUP AND SAUCER มี 70.00%

## 3.ของช่วงเวลาตรุษจีนในทวีป Asia ( 15 Jan - 15 Feb)

In [33]:
Asia = data[(data['Country']=='Bahrain') | (data['Country']=='Hong Kong')| (data['Country']=='Singapore') | (data['Country']=='Japan') | (data['Country']=='Israel') |
            (data['Country']=='Lebanon') | (data['Country']=='Saudi Arabia') | (data['Country']=='United Arab Emirates')]
Asia #แบ่งทวีปเอเชีย

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                     Description  Quantity  \
9783      537218     85016   SET OF 6 VINTAGE NOTELETS KIT         6   
9784      537218     21506      FANCY FONT BIRTHDAY CARD,         12   
9785      537218     22716              CARD CIRCUS PARADE        12   
9786      537218     22983             CARD BILLBOARD FONT        12   
9787      537218     22030          SWALLOWS GREETING CARD        12   
...          ...       ...                             ...       ...   
530276   C580832     21731   RED TOADSTOOL LED NIGHT LIGHT        -2   
530277   C580832     23480  MINI LIGHTS WOODLAND MUSHROOMS       -15   
530278   C580832     22466  FAIRY TALE COTTAGE NIGHT LIGHT       -17   
530279   C580832     23084              RABBIT NIGHT LIGHT        -7   
530280   C580832     22621      TRADITIONAL KNITTING NANCY        -1   

               InvoiceDate  UnitPrice  CustomerID Country       Date  Month  
9783   2010-12-05 15:46:00       2.55     12763.0   Japan 2010-12-05     12  
9784   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9785   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9786   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
9787   2010-12-05 15:46:00       0.42     12763.0   Japan 2010-12-05     12  
...                    ...        ...         ...     ...        ...    ...  
530276 2011-12-06 11:40:00       1.45     12753.0   Japan 2011-12-06     12  
530277 2011-12-06 11:40:00       3.29     12753.0   Japan 2011-12-06     12  
530278 2011-12-06 11:40:00       1.65     12753.0   Japan 2011-12-06     12  
530279 2011-12-06 11:40:00       1.79     12753.0   Japan 2011-12-06     12  
530280 2011-12-06 11:40:00       1.45     12753.0   Japan 2011-12-06     12  

[1314 rows x 10 columns]

In [34]:
#Chinese Newyear
China = Asia[(Asia['Date'].dt.month == 1) & (Asia['Date'].dt.day >= 15) & (data['Date'].dt.day <= 31)  |
                   ((Asia['Date'].dt.month == 2) & (Asia['Date'].dt.day <= 15))]
#print check
print(China)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


      InvoiceNo StockCode                          Description  Quantity  \
69623    541975     22620          4 TRADITIONAL SPINNING TOPS        60   
69624    541975    85183B  CHARLIE & LOLA WASTEPAPER BIN FLORA         6   
69625    541975    85183A   CHARLIE & LOLA WASTEPAPER BIN BLUE         6   
69626    541975     16236                 KITTY PENCIL ERASERS        28   
69627    541975     21058               PARTY INVITES WOODLAND        12   
...         ...       ...                                  ...       ...   
89595    543911    47590B          PINK HAPPY BIRTHDAY BUNTING        12   
89596    543911     22117     METAL SIGN HER DINNER IS SERVED         12   
89597    543911     82567            AIRLINE LOUNGE,METAL SIGN        12   
89598    543911     21910                  WAY OUT METAL SIGN         12   
89599    543911     21911                   GARDEN METAL SIGN         12   

              InvoiceDate  UnitPrice  CustomerID               Country  \
69623 2011-01

<ipython-input-34-0af9221392ec>:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  China = Asia[(Asia['Date'].dt.month == 1) & (Asia['Date'].dt.day >= 15) & (data['Date'].dt.day <= 31)  |


In [35]:
trans_chin = China.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans_chin_apply = trans_chin['Description'].apply(lambda x: [str(item) for item in x])

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [36]:
trans_chin_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0    [4 TRADITIONAL SPINNING TOPS, CHARLIE & LOLA W...
1    [GIN & TONIC DIET GREETING CARD , GREEN REGENC...
2    [LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT MET...
3    [MONEY BOX FIRST ADE DESIGN, MONEY BOX KINGS C...
4    [SET OF 3 NOTEBOOKS IN PARCEL, LUNCH BAG DOLLY...
5    [RETROSPOT HEART HOT WATER BOTTLE, DOORMAT UNI...
Name: Description, dtype: object

In [37]:
min_sup = 0.33 #เปลี่ยนจนกว่าจะเจอ...

frequentItemset_China = list(apriori(trans_chin_apply, min_support = min_sup))
for i in frequentItemset_China:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'LUNCH BAG RED RETROSPOT', 'BASKET OF TOADSTOOLS'}), support=0.3333333333333333, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'LUNCH BAG RED RETROSPOT', 'BASKET OF TOADSTOOLS'}), confidence=0.3333333333333333, lift=1.0), OrderedStatistic(items_base=frozenset({'BASKET OF TOADSTOOLS'}), items_add=frozenset({'LUNCH BAG RED RETROSPOT'}), confidence=1.0, lift=3.0), OrderedStatistic(items_base=frozenset({'LUNCH BAG RED RETROSPOT'}), items_add=frozenset({'BASKET OF TOADSTOOLS'}), confidence=1.0, lift=3.0)])
yes
RelationRecord(items=frozenset({'SET 3 RETROSPOT TEA,COFFEE,SUGAR', 'BASKET OF TOADSTOOLS'}), support=0.3333333333333333, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'BASKET OF TOADSTOOLS', 'SET 3 RETROSPOT TEA,COFFEE,SUGAR'}), confidence=0.3333333333333333, lift=1.0), OrderedStatistic(items_base=frozenset({'BASKET OF TOADSTOOLS'}), items_add=frozenset({'SET 3 RETROSPOT TEA,CO

In [38]:
frequentItemset_China = list(apriori(trans_chin_apply, min_support=0.33))

for record in frequentItemset_China:
    if len(record.items) > 1:
        print('yes')
        print(f"Items: {', '.join(record.items)}")
        print(f"Support: {record.support:.4f} ({record.support*100:.2f}%)")

        for i, stat in enumerate(record.ordered_statistics, 1):
            items_base = 'None' if not stat.items_base else ', '.join(stat.items_base)
            items_add = ', '.join(stat.items_add)
            confidence = stat.confidence
            lift = stat.lift

            print(f"Rule {i}: ({items_base} -> {items_add} : Confidence={confidence:.4f}, Lift={lift:.4f})")
        print()

yes
Items: LUNCH BAG RED RETROSPOT, BASKET OF TOADSTOOLS
Support: 0.3333 (33.33%)
Rule 1: (None -> LUNCH BAG RED RETROSPOT, BASKET OF TOADSTOOLS : Confidence=0.3333, Lift=1.0000)
Rule 2: (BASKET OF TOADSTOOLS -> LUNCH BAG RED RETROSPOT : Confidence=1.0000, Lift=3.0000)
Rule 3: (LUNCH BAG RED RETROSPOT -> BASKET OF TOADSTOOLS : Confidence=1.0000, Lift=3.0000)

yes
Items: SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS
Support: 0.3333 (33.33%)
Rule 1: (None -> BASKET OF TOADSTOOLS, SET 3 RETROSPOT TEA,COFFEE,SUGAR : Confidence=0.3333, Lift=1.0000)
Rule 2: (BASKET OF TOADSTOOLS -> SET 3 RETROSPOT TEA,COFFEE,SUGAR : Confidence=1.0000, Lift=3.0000)
Rule 3: (SET 3 RETROSPOT TEA,COFFEE,SUGAR -> BASKET OF TOADSTOOLS : Confidence=1.0000, Lift=3.0000)

yes
Items: JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD
Support: 0.3333 (33.33%)
Rule 1: (None -> JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD : Confidence=0.3333, Lift=1.0000)
Rule 2: (BATH BUILDING BLOCK WORD -> JAM MAKING SET

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 2 itemset ที่ผ่าน 0.03

1. คนที่ซื้อทั้ง BASKET OF TOADSTOOLS และ LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

   คนที่ซื้อ BASKET OF TOADSTOOLS และ LUNCH BAG RED RETROSPOT มี 33.33%

   คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

   คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

2. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR และ BASKET OF TOADSTOOLS จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ RETROSPOT TEA,COFFEE,SUGAR และ BASKET OF TOADSTOOLS มี 33.33%

  คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

3. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ SET 3 JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

4. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ SET 3 LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 66.67%

--------------------------------------------------------------------------------

5. คนที่ซื้อทั้ง SET 3 PAPER VINTAGE CHICK PAPER EGG และ BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 66.67%

--------------------------------------------------------------------------------

6. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 66.67%

--------------------------------------------------------------------------------

7. คนที่ซื้อทั้ง LAUNDRY 15C METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

8. คนที่ซื้อทั้ง WASHROOM METAL SIGN และ BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN และ BEWARE OF THE CAT METAL SIGN มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

9. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ DOORMAT UNION FLAG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ DOORMAT UNION FLAG มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ DOORMAT UNION FLAG มี 66.67%

--------------------------------------------------------------------------------

10. คนที่ซื้อทั้ง DOORMAT UNION FLAG และ METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG และ METAL SIGN TAKE IT OR LEAVE IT มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT มี 100.00%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT แล้วค่อยซื้อ DOORMAT UNION FLAG มี 100.00%

--------------------------------------------------------------------------------

11. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ JAM MAKING SET WITH JARS จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ JAM MAKING SET WITH JARS มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 66.67%

--------------------------------------------------------------------------------

12. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 66.67%

--------------------------------------------------------------------------------

13. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ LAUNDRY 15C METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ LAUNDRY 15C METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

14. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ METAL SIGN TAKE IT OR LEAVE IT มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT มี 66.67%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

15. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 66.67%

--------------------------------------------------------------------------------

16. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN และ WASHROOM METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN และ WASHROOM METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 66.67%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

17. คนที่ซื้อทั้ง WASHROOM METAL SIGN และ LAUNDRY 15C METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN และ LAUNDRY 15C METAL SIGN มี 33.33%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

18. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR และ LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR และ LUNCH BAG RED RETROSPOT มี 33.33%

  คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

19. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ MILK PAN RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ MILK PAN RED RETROSPOT มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

20. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

21. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT และ RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT และ RED RETROSPOT CHILDRENS UMBRELLA ทั้งหมด มี 33.33% มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

22. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT และ PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT และ PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

23. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT และ RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT และ RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

24. คนที่ซื้อทั้ง PARTY BUNTING และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ PARTY BUNTING และ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ PARTY BUNTING แล้วค่อยซื้อ PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ PARTY BUNTING มี 66.67%

--------------------------------------------------------------------------------

25. คนที่ซื้อทั้ง PLASTERS IN TIN SPACEBOY และ RED RETROSPOT CHILDRENS UMBRELLA EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY และ RED RETROSPOT CHILDRENS UMBRELLA EGG มี 33.33%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA EGG มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA EGG แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

--------------------------------------------------------------------------------

26. คนที่ซื้อทั้ง SET OF 72 PINK HEART PAPER DOILIES และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ SET OF 72 PINK HEART PAPER DOILIES มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ SET OF 72 PINK HEART PAPER DOILIES มี 66.67%

  คนที่ซื้อ SET OF 72 PINK HEART PAPER DOILIES แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

--------------------------------------------------------------------------------

27. คนที่ซื้อทั้ง SET OF 72 PINK HEART PAPER DOILIES และ SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG และ SET OF 72 PINK HEART PAPER DOILIES มี 33.33%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ SET OF 72 PINK HEART PAPER DOILIES มี 66.67%

  คนที่ซื้อ SET OF 72 PINK HEART PAPER DOILIES แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

--------------------------------------------------------------------------------

28. คนที่ซื้อทั้ง SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT มี 33.33%

  คนที่ซื้อ BASKET OF TOADSTOOLS แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR แล้วค่อยซื้อ BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ BASKET OF TOADSTOOLS, LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, BASKET OF TOADSTOOLS แล้วค่อยซื้อ LUNCH BAG RED RETROSPOT มี 100.00%

  คนที่ซื้อ SET 3 RETROSPOT TEA,COFFEE,SUGAR, LUNCH BAG RED RETROSPOT แล้วค่อยซื้อ BASKET OF TOADSTOOLS มี 100.00%

--------------------------------------------------------------------------------

29. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

30. คนที่ซื้อทั้ง JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

31. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

32. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

33. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

34. คนที่ซื้อทั้ง WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

35. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  มี 33.33%

  คนที่ซื้อ DOORMAT UNION FLAG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, METAL SIGN TAKE IT OR LEAVE IT  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  มี 66.67%

  คนที่ซื้อ METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, DOORMAT UNION FLAG แล้วค่อยซื้อ METAL SIGN TAKE IT OR LEAVE IT  มี 100.00%

  คนที่ซื้อ DOORMAT UNION FLAG, METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, METAL SIGN TAKE IT OR LEAVE IT  แล้วค่อยซื้อ DOORMAT UNION FLAG มี 100.00%

--------------------------------------------------------------------------------

36. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 33.33%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 66.67%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

--------------------------------------------------------------------------------

37. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

--------------------------------------------------------------------------------

38. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

39. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

40. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

--------------------------------------------------------------------------------

41. คนที่ซื้อทั้ง MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

--------------------------------------------------------------------------------

42. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD มี 33.33%

  คนที่ซื้อ BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 66.67%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 66.67%

  คนที่ซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS แล้วค่อยซื้อ BATH BUILDING BLOCK WORD, SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ SET 3 PAPER VINTAGE CHICK PAPER EGG มี 100.00%

  คนที่ซื้อ JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, SET 3 PAPER VINTAGE CHICK PAPER EGG, BATH BUILDING BLOCK WORD แล้วค่อยซื้อ JAM MAKING SET WITH JARS มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, JAM MAKING SET WITH JARS, SET 3 PAPER VINTAGE CHICK PAPER EGG แล้วค่อยซื้อ BATH BUILDING BLOCK WORD มี 100.00%

--------------------------------------------------------------------------------

43. คนที่ซื้อทั้ง LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 33.33%

  คนที่ซื้อ BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 66.67%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN, BEWARE OF THE CAT METAL SIGN  มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ WASHROOM METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, WASHROOM METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LAUNDRY 15C METAL SIGN มี 100.00%

  คนที่ซื้อ WASHROOM METAL SIGN, LAUNDRY 15C METAL SIGN, BEWARE OF THE CAT METAL SIGN  แล้วค่อยซื้อ LADIES & GENTLEMEN METAL SIGN มี 100.00%

  คนที่ซื้อ LADIES & GENTLEMEN METAL SIGN, LAUNDRY 15C METAL SIGN, WASHROOM METAL SIGN แล้วค่อยซื้อ BEWARE OF THE CAT METAL SIGN  มี 100.00%

--------------------------------------------------------------------------------

44. คนที่ซื้อทั้ง MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA จาก dataset ทั้งหมด มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 33.33%

  คนที่ซื้อ MILK PAN PINK POLKADOT แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ MILK PAN PINK POLKADOT, RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY แล้วค่อยซื้อ RED RETROSPOT CHILDRENS UMBRELLA มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, MILK PAN RED RETROSPOT, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ PLASTERS IN TIN SPACEBOY มี 100.00%

  คนที่ซื้อ MILK PAN PINK POLKADOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN RED RETROSPOT มี 100.00%

  คนที่ซื้อ MILK PAN RED RETROSPOT, PLASTERS IN TIN SPACEBOY, RED RETROSPOT CHILDRENS UMBRELLA แล้วค่อยซื้อ MILK PAN PINK POLKADOT มี 100.00%

## 4. ของช่วงเวลา Christmas & New year Europe (1 dec - 30 dec)


In [39]:
Eur

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[541909 rows x 10 columns]

In [40]:
#Select 1-31 Dec
Dec_2 = Eur[(Eur['Date'].dt.month == 12) & (Eur['Date'].dt.day >= 1) & (Eur['Date'].dt.day <= 31)]
#print check
print(Dec_2)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


       InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Dat

In [41]:
#EU Christmas
Eu_xmas = Eur[(Eur['Date'].dt.month == 12) & (Eur['Date'].dt.day >= 1) & (Eur['Date'].dt.day <= 31)]
Eu_xmas

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


InvoiceNo StockCode                          Description  Quantity  \
0         536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1         536365     71053                  WHITE METAL LANTERN         6   
2         536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3         536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4         536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
...          ...       ...                                  ...       ...   
541904    581587     22613          PACK OF 20 SPACEBOY NAPKINS        12   
541905    581587     22899         CHILDREN'S APRON DOLLY GIRL          6   
541906    581587     23254        CHILDRENS CUTLERY DOLLY GIRL          4   
541907    581587     23255      CHILDRENS CUTLERY CIRCUS PARADE         4   
541908    581587     22138        BAKING SET 9 PIECE RETROSPOT          3   

               InvoiceDate  UnitPrice  CustomerID         Country       Date  \
0      2010-12-01 08:26:00       2.55     17850.0  United Kingdom 2010-12-01   
1      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
2      2010-12-01 08:26:00       2.75     17850.0  United Kingdom 2010-12-01   
3      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
4      2010-12-01 08:26:00       3.39     17850.0  United Kingdom 2010-12-01   
...                    ...        ...         ...             ...        ...   
541904 2011-12-09 12:50:00       0.85     12680.0          France 2011-12-09   
541905 2011-12-09 12:50:00       2.10     12680.0          France 2011-12-09   
541906 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541907 2011-12-09 12:50:00       4.15     12680.0          France 2011-12-09   
541908 2011-12-09 12:50:00       4.95     12680.0          France 2011-12-09   

        Month  
0          12  
1          12  
2          12  
3          12  
4          12  
...       ...  
541904     12  
541905     12  
541906     12  
541907     12  
541908     12  

[68006 rows x 10 columns]

In [42]:
trans_xmas = Eu_xmas.groupby('InvoiceNo')['Description'].apply(list).reset_index()
trans_xmas_apply = trans_xmas['Description'].apply(lambda x: [str(item) for item in x])
trans_xmas_apply

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0       [WHITE HANGING HEART T-LIGHT HOLDER, WHITE MET...
1       [HAND WARMER UNION JACK, HAND WARMER RED POLKA...
2       [ASSORTED COLOUR BIRD ORNAMENT, POPPY'S PLAYHO...
3       [JAM MAKING SET WITH JARS, RED COAT RACK PARIS...
4                              [BATH BUILDING BLOCK WORD]
                              ...                        
3035                        [PAPER CRAFT , LITTLE BIRDIE]
3036    [VICTORIAN GLASS HANGING T-LIGHT, ZINC T-LIGHT...
3037                                             [Manual]
3038                         [VICTORIAN SEWING BOX LARGE]
3039    [HANGING HEART JAR T-LIGHT HOLDER, 36 PENCILS ...
Name: Description, Length: 3040, dtype: object

In [43]:
min_sup = 0.036 #เปลี่ยนจนกว่าจะเจอ...

frequentItemset_xmas = list(apriori(trans_xmas_apply, min_support = min_sup))
for i in frequentItemset_xmas:
    if i[0].__len__()>1:
        print('yes')
        print(i)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


yes
RelationRecord(items=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), support=0.03684210526315789, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN', 'ALARM CLOCK BAKELIKE RED '}), confidence=0.03684210526315789, lift=1.0), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), items_add=frozenset({'ALARM CLOCK BAKELIKE RED '}), confidence=0.7088607594936708, lift=14.462662475575565), OrderedStatistic(items_base=frozenset({'ALARM CLOCK BAKELIKE RED '}), items_add=frozenset({'ALARM CLOCK BAKELIKE GREEN'}), confidence=0.7516778523489932, lift=14.462662475575565)])
yes
RelationRecord(items=frozenset({'CHOCOLATE HOT WATER BOTTLE', 'HOT WATER BOTTLE I AM SO POORLY'}), support=0.036513157894736845, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'CHOCOLATE HOT WATER BOTTLE', 'HOT WATER BOTTLE I AM SO POORLY'}), confidence=0.036513157894736845, lift=1.0), O

In [44]:
frequentItemset_xmas = list(apriori(trans_xmas_apply, min_support=0.035))

for record in frequentItemset_xmas:
    if len(record.items) > 1:
        print('yes')
        print(f"Items: {', '.join(record.items)}")
        print(f"Support: {record.support:.4f} ({record.support*100:.2f}%)")

        for i, stat in enumerate(record.ordered_statistics, 1):
            items_base = 'None' if not stat.items_base else ', '.join(stat.items_base)
            items_add = ', '.join(stat.items_add)
            confidence = stat.confidence
            lift = stat.lift

            print(f"Rule {i}: ({items_base} -> {items_add} : Confidence={confidence:.4f}, Lift={lift:.4f})")
        print()

yes
Items: ALARM CLOCK BAKELIKE GREEN, ALARM CLOCK BAKELIKE RED 
Support: 0.0368 (3.68%)
Rule 1: (None -> ALARM CLOCK BAKELIKE GREEN, ALARM CLOCK BAKELIKE RED  : Confidence=0.0368, Lift=1.0000)
Rule 2: (ALARM CLOCK BAKELIKE GREEN -> ALARM CLOCK BAKELIKE RED  : Confidence=0.7089, Lift=14.4627)
Rule 3: (ALARM CLOCK BAKELIKE RED  -> ALARM CLOCK BAKELIKE GREEN : Confidence=0.7517, Lift=14.4627)

yes
Items: CHOCOLATE HOT WATER BOTTLE, HOT WATER BOTTLE I AM SO POORLY
Support: 0.0365 (3.65%)
Rule 1: (None -> CHOCOLATE HOT WATER BOTTLE, HOT WATER BOTTLE I AM SO POORLY : Confidence=0.0365, Lift=1.0000)
Rule 2: (CHOCOLATE HOT WATER BOTTLE -> HOT WATER BOTTLE I AM SO POORLY : Confidence=0.5163, Lift=9.7484)
Rule 3: (HOT WATER BOTTLE I AM SO POORLY -> CHOCOLATE HOT WATER BOTTLE : Confidence=0.6894, Lift=9.7484)

yes
Items: PAPER CHAIN KIT VINTAGE CHRISTMAS, PAPER CHAIN KIT 50'S CHRISTMAS 
Support: 0.0477 (4.77%)
Rule 1: (None -> PAPER CHAIN KIT 50'S CHRISTMAS , PAPER CHAIN KIT VINTAGE CHRISTMAS : 

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 3 itemset ที่ผ่าน 0.036
1. คนที่ซื้อทั้ง PAPER CHAIN KIT 50'S CHRISTMAS และ PAPER CHAIN KIT VINTAGE CHRISTMAS จาก dataset ทั้งหมด มี 4.77%

   คนที่ซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS และPAPER CHAIN KIT 50'S CHRISTMAS  มี 4.77%
   
   คนที่ซื้อ PAPER CHAIN KIT 50'S CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS มี 51.6%

   คนที่ซื้อ PAPER CHAIN KIT VINTAGE CHRISTMAS แล้วค่อยซื้อ PAPER CHAIN KIT 50'S CHRISTMAS มี 66.51%


2. คนที่ซื้อทั้ง ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN จาก dataset ทั้งหมด มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED และ ALARM CLOCK BAKELIKE GREEN มี 3.68%

   คนที่ซื้อ ALARM CLOCK BAKELIKE GREEN แล้วค่อยซื้อ ALARM CLOCK BAKELIKE RED มี 75.17%

   คนที่ซื้อ ALARM CLOCK BAKELIKE RED แล้วค่อยซื้อ ALARM CLOCK BAKELIKE GREEN มี 70.89%



3. คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY จาก dataset ทั้งหมด มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE และ HOT WATER BOTTLE I AM SO POORLY มี 3.65%

   คนที่ซื้อ CHOCOLATE HOT WATER BOTTLE แล้วค่อยซื้อ HOT WATER BOTTLE I AM SO POORLY มี 51.63%

   คนที่ซื้อ HOT WATER BOTTLE I AM SO POORLY แล้วค่อยซื้อ CHOCOLATE HOT WATER BOTTLE มี 68.94%


## 5. Valentine's day


/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


## 6. โดยจัดกลุ่มตามประเทศ groupby ประเทศ - เดือน

In [45]:
trans_CM = data.groupby(['Country','Month'])['Description'].apply(list).reset_index()
trans_Coun_M = trans_CM['Description'].apply(lambda x: [str(item) for item in x])
trans_Coun_M

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


0      [RIBBON REEL LACE DESIGN , RIBBON REEL HEARTS ...
1      [HANGING HEART JAR T-LIGHT HOLDER, JAM MAKING ...
2      [PANTRY ROLLING PIN, PANTRY SCRUBBING BRUSH, S...
3      [POSTAGE, JIGSAW TOADSTOOLS 3 PIECE, SET OF 4 ...
4      [ROSES REGENCY TEACUP AND SAUCER , EASTER TIN ...
                             ...                        
291    [PARTY CHARMS 50 PIECES, PACK OF 6 LARGE FRUIT...
292    [ASSORTED MINI MADRAS NOTEBOOK, PACK OF 72 RET...
293    [STORAGE TIN VINTAGE DOILY , SET OF 72 RETROSP...
294    [CALENDAR PAPER CUT DESIGN, DINOSAURS  WRITING...
295    [ROUND SNACK BOXES SET OF4 WOODLAND , LUNCH BA...
Name: Description, Length: 296, dtype: object

In [46]:
min_sup = 0.32 #เปลี่ยนจนกว่าจะเจอ...

frequentItemset_CM = list(apriori(trans_Coun_M, min_support = min_sup))
for i in frequentItemset_CM:
    if i[0].__len__()>1:
        print('yes')
        print(i)

yes
RelationRecord(items=frozenset({'POSTAGE', 'ROUND SNACK BOXES SET OF4 WOODLAND '}), support=0.32094594594594594, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'POSTAGE', 'ROUND SNACK BOXES SET OF4 WOODLAND '}), confidence=0.32094594594594594, lift=1.0), OrderedStatistic(items_base=frozenset({'POSTAGE'}), items_add=frozenset({'ROUND SNACK BOXES SET OF4 WOODLAND '}), confidence=0.5588235294117647, lift=1.463820926600729), OrderedStatistic(items_base=frozenset({'ROUND SNACK BOXES SET OF4 WOODLAND '}), items_add=frozenset({'POSTAGE'}), confidence=0.8407079646017699, lift=1.4638209266007287)])


/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


In [47]:
frequentItemset_CM = list(apriori(trans_Coun_M, min_support=0.32))

for record in frequentItemset_CM:
    if len(record.items) > 1:
        print('yes')
        print(f"Items: {', '.join(record.items)}")
        print(f"Support: {record.support:.4f} ({record.support*100:.2f}%)")

        for i, stat in enumerate(record.ordered_statistics, 1):
            items_base = 'None' if not stat.items_base else ', '.join(stat.items_base)
            items_add = ', '.join(stat.items_add)
            confidence = stat.confidence
            lift = stat.lift

            print(f"Rule {i}: ({items_base} -> {items_add} : Confidence={confidence:.4f}, Lift={lift:.4f})")
        print()

yes
Items: POSTAGE, ROUND SNACK BOXES SET OF4 WOODLAND 
Support: 0.3209 (32.09%)
Rule 1: (None -> POSTAGE, ROUND SNACK BOXES SET OF4 WOODLAND  : Confidence=0.3209, Lift=1.0000)
Rule 2: (POSTAGE -> ROUND SNACK BOXES SET OF4 WOODLAND  : Confidence=0.5588, Lift=1.4638)
Rule 3: (ROUND SNACK BOXES SET OF4 WOODLAND  -> POSTAGE : Confidence=0.8407, Lift=1.4638)



/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


มี 1 itemset ที่ผ่าน 0.32
1. คนที่ซื้อทั้ง POSTAGE และ ROUND SNACK BOXES SET OF4 WOODLAND จาก dataset ทั้งหมด มี 32.09%

   คนที่ซื้อ POSTAGE และ ROUND SNACK BOXES SET OF4 WOODLAN มี 32.09%
  
   คนที่ซื้อ POSTAGE แล้วค่อยซื้อ ROUND SNACK BOXES SET OF4 WOODLAND มี 55.88%

   คนที่ซื้อ ROUND SNACK BOXES SET OF4 WOODLAND แล้วค่อยซื้อ POSTAGE มี 84.07%
